feat(gradle): support useJUnitJupiter under suites.withType

trask opened the pull request:
## Changes

Extends the support for `useJUnitJupiter("<version>")` to also cover this usage:

```
testing {
  suites.withType(JvmTestSuite::class).configureEach {
    useJUnitJupiter("<version>")
  }
}
```

## Context

Please select one of the below:

- [ ] This closes an existing Issue: #
- [x] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.

- [ ] No — I did not use AI for this contribution.
- [ ] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [x] Yes — substantive assistance (AI generated non‑trivial portions of code, tests, or documentation).
- [ ] Yes — other (please describe):

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

The public repository: https://github.com/trask/opentelemetry-java-contrib-renovate

PR it generated: https://github.com/trask/opentelemetry-java-contrib-renovate/pull/3

trask committed:
Support useJUnitJupiter under suites.withType

trask committed:
unify

RahulGautamSingh reviewed (approved):
LGTM

Churro reviewed (changes requested):
The approach is completely reasonable but it is possible to avoid redundancy and shorten these changes by simply integrating the `withType` parts into `qImplicitTestSuites` using `q.alt`:

```ts
// testing { suites { test { useSpock("1.2.3") } } }
// testing.suites.test { useJunit("1.2.3") }
// testing { suites.withType(JvmTestSuite).configureEach { useJUnitJupiter("5.13.4") } }
// testing { suites.withType(JvmTestSuite::class).configureEach { useSpock("2.3") } }
const qImplicitTestSuites = qDotOrBraceExpr(
  'testing',
  qDotOrBraceExpr(
    'suites',
    q.alt(
      qDotOrBraceExpr('test', qTestSuiteMethod),
      q
        .sym<Ctx>('withType')
        .tree({
          type: 'wrapped-tree',
          maxDepth: 1,
          startsWith: '(',
          endsWith: ')',
        })
        .op('.')
        .sym('configureEach')
        .tree({
          type: 'wrapped-tree',
          maxDepth: 1,
          startsWith: '{',
          endsWith: '}',
          search: qTestSuiteMethod,
        }),
    ),
  ),
)
  .handler(handleImplicitDep)
  .handler(cleanupTempVars);
```

viceice reviewed (commented):
see open conversation

trask committed:
fix: change withType maxDepth from 2 to 1

Addresses review comment from @Churro about maxDepth: 2 being too permissive.
With maxDepth: 1, the withType tree will only match:
- withType(JvmTestSuite)
- withType(JvmTestSuite::class)

This prevents unintended matching of nested structures like withType([...])
which are not valid Gradle syntax for this use case and are not covered by tests.

trask committed:
refactor: merge test suite queries using q.alt pattern

trask commented:
> The approach is completely reasonable but it is possible to avoid redundancy and shorten these changes by simply integrating the `withType` parts into `qImplicitTestSuites` using `q.alt`

applied and retested (https://github.com/trask/opentelemetry-java-contrib-renovate/pull/7)
